### PR TITLE
fix: Search button mobile

### PR DESCRIPTION
**Pull request by WaDadidou**

### CODE REVIEW

File: packages/context/SearchBarProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import { selectSearchText } from "../store/slices/search";
+import { getMarketplaceClient } from "../utils/backend";
+
+interface DefaultValue {
+  isSearchModalMobileOpen: boolean;
+  setSearchModalMobileOpen: (value: boolean) => void;
+  names?: string[];
+  collections?: Collection[];
+  hasCollections?: boolean;
+  hasNames?: boolean;
```
Comment (n0izn0iz) [inline packages/context/SearchBarProvider.tsx:16]:
```suggestion
  names: string[];
  collections: Collection[];
  hasCollections: boolean;
  hasNames: boolean;
```

we can provide sane default values for these so I recommend to not allow them to be undefined

**Review by n0izn0iz — CHANGES_REQUESTED** [review]

File: packages/context/SearchBarProvider.tsx (hunk 1 above)
Comment (WaDadidou) [reply packages/context/SearchBarProvider.tsx:16]: Fixed

**Comment by n0izn0iz** [comment]
also the search is opened on page load on mobile

**Review by n0izn0iz — APPROVED** [review]
LGTM, please fix lint warning

File: packages/components/Search/SearchBarInput.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          {
+            color: "white",
+            flex: 1,
+            marginLeft: 8,
+          },
+          { outlineStyle: "none" } as any,
+        ]}
+        onChangeText={(text) => {
+          dispatch(setSearchText(text));
+          onInteraction && onInteraction();
```
Comment (hthieu1110) [inline packages/components/Search/SearchBarInput.tsx:48]: Shortcut: onInteraction?.();

**Review by hthieu1110 — APPROVED** [review]
LGTM

**Pull request merged**